src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        int hash = hash(key);
+        int index = (table.length - 1) & hash;
```

Comment:
See comment about `calculateIndex` method, now you can do it in one line
```suggestion
        int index = calculateIndex(key);
```